strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/AbstractLayoutProvider.java

```diff
@@ -63,4 +79,132 @@ public Storage getStorage(String storageId)
         return configurationManager.getConfiguration().getStorage(storageId);
     }
 
+    @Override
+    public ArtifactInputStream getInputStream(String storageId,
+                                              String repositoryId,
+                                              String path)
+            throws IOException, NoSuchAlgorithmException
+    {
+        Storage storage = getConfiguration().getStorage(storageId);
+
+        logger.debug("Checking in " + storage.getId() + ":" + repositoryId + "...");
+
+        Repository repository = storage.getRepository(repositoryId);
+        StorageProvider storageProvider = storageProviderRegistry.getProvider(repository.getImplementation());
+
+        ArtifactInputStream ais;
+        if (isArtifact(repository, path, true))
+        {
+            ArtifactPath artifactPath = resolve(repository, getArtifactCoordinates(path));
+            ais = storageProvider.getInputStreamImplementation(artifactPath);
+        } else
+        {
+            RepositoryPath repositoryPath = resolve(repository);
+            ais = storageProvider.getInputStreamImplementation(repositoryPath, path);
+        }
+        logger.debug("Resolved " + path + "!");
+        return ais;
+    }
+
+    @Override
+    public ArtifactOutputStream getOutputStream(String storageId,
+                                                String repositoryId,
+                                                String path)
+        throws IOException
+    {
+        Storage storage = getConfiguration().getStorage(storageId);
+        Repository repository = storage.getRepository(repositoryId);
+        StorageProvider storageProvider = storageProviderRegistry.getProvider(repository.getImplementation());
+        OutputStream os;
+        if (isArtifact(repository, path, false))
+        {
+            ArtifactPath artifactPath = resolve(repository, getArtifactCoordinates(path));
+            os = storageProvider.getOutputStreamImplementation(artifactPath);
+        } else
+        {
+            RepositoryPath repositoryPath = resolve(repository);
+            os = storageProvider.getOutputStreamImplementation(repositoryPath, path);
+        }
+
+        return os instanceof ArtifactOutputStream ? (ArtifactOutputStream) os : new ArtifactOutputStream(os, null);
+    }
+    
+    protected abstract boolean isMetadata(String path);
+    
+    protected abstract boolean isChecksum(String path);
+    
+    protected boolean isTrash(String path){
```

Comment:
Cosmetic: Could you please sort out all the curly braces to match the project's coding convention and be on the next line? (Could you please have a look at the rest of your code for occurrences of  `) {` and sort them out...?). Thanks! :)